### PR TITLE
Make docker default registry optional

### LINKED ISSUE #16: Make default docker registry installation configurable
Make the installation of the default docker registry at startup configurable by adding a registry key.

### PR DESCRIPTION
**Pull request by robertauer**
Only install default docker registry if installDefaultDockerRegistry key is set to true

resolves #16

### CODE REVIEW

File: resources/startup.sh [hunk 1]
```diff
@@ -142,9 +148,18 @@ if [ "$(doguctl config successfulInitialConfiguration)" != "true" ]; then
   exportNexusPassword
 
   export NEXUS_SERVER="http://localhost:8081/nexus"
-  nexus-claim plan -i /defaultDockerRegistry.hcl -o "-" | nexus-claim apply -i "-"
+
+  # Install default docker registry if requested via etcd
+  if doguctl config installDefaultDockerRegistry > /dev/null ; then
```
Comment (mbehlendorf) [inline resources/startup.sh]: Is the double-if really necessary? Maybe we could work with one statement, an & or the --default parameter of doguctl.

File: resources/startup.sh [hunk 2]
```diff
@@ -142,9 +148,18 @@ if [ "$(doguctl config successfulInitialConfiguration)" != "true" ]; then
   exportNexusPassword
 
   export NEXUS_SERVER="http://localhost:8081/nexus"
```
Comment (mbehlendorf) [inline resources/startup.sh]: This should be part of the installDefaultDockerRegistry function. Maybe you could also use the -s parameter of nexus-claim

**Comment by mbehlendorf** [comment]
Is it correct that the additional service is configured even if the default docker registry is not installed? https://github.com/cloudogu/nexus/blob/a11a6bcc8966175a8b2b46fce86f15ca7f222ed7/Dockerfile#L14

**Comment by robertauer** [comment]
> Is it correct that the additional service is configured even if the default docker registry is not installed?
> 
> [nexus/Dockerfile](https://github.com/cloudogu/nexus/blob/a11a6bcc8966175a8b2b46fce86f15ca7f222ed7/Dockerfile#L14)
> 
> Line 14 in [a11a6bc](/cloudogu/nexus/commit/a11a6bcc8966175a8b2b46fce86f15ca7f222ed7)
>  SERVICE_ADDITIONAL_SERVICES='[{"name": "docker-registry", "location": "v2", "pass": "nexus/repository/docker-registry/v2/"}]' \

Yes, because the docker registry can possibly be added manually afterwards, which can hardly be noticed at runtime.

**Pull request merged**